### PR TITLE
Foam Rebalance

### PR DESCRIPTION
**Pull request by Iamgoofball**
:cl: Iamgoofball
fix: Foam no longer duplicates reagents times four. What you put in is what you get out.
balance: Foam no longer runs out of reagents.
balance: Foam uses TOUCH instead of Vapor
balance: Fire dissolving foam was removed due to never occuring and being code bloat
code: Speeds up foam code by a lot theoretically. Lots of dangerous foam shouldn't be nearly as laggy now.
fix: Foam now lasts exactly 10 seconds, instead of the previously random length due to processing speed varying.
/:cl:

This kills foam being super omega deadly and helps speed it up while stabilizing its behavior.

### CODE REVIEW

**Comment by Iamgoofball** [comment]
DNM need to migrate over to old loop system

File: code/game/objects/effects/effect_system/effects_foam.dm [hunk 1]
```diff
@@ -66,9 +65,6 @@
 	L.adjust_fire_stacks(-2)
 	L.ExtinguishMob()
 
-/obj/effect/particle_effect/foam/firefighting/temperature_expose(datum/gas_mixture/air, exposed_temperature, exposed_volume)
-	return
```
Comment (DaedalusGame) [inline code/game/objects/effects/effect_system/effects_foam.dm]: This is to stop fire from deleting it instantly.

File: code/game/objects/effects/effect_system/effects_foam.dm [hunk 2]
```diff
@@ -233,20 +217,27 @@
 	else
 		location = get_turf(loca)
 
-	amount = round(sqrt(amt / 2), 1)
-	carry.copy_to(chemholder, 4*carry.total_volume) //The foam holds 4 times the total reagents volume for balance purposes.
+	carry.copy_to(chemholder, carry.total_volume)
```
Comment (DaedalusGame) [inline code/game/objects/effects/effect_system/effects_foam.dm]: Thank fucking god this is actually correct for once

File: code/game/objects/effects/effect_system/effects_foam.dm [hunk 3]
```diff
@@ -122,80 +119,67 @@
...
-		var/obj/effect/particle_effect/foam/F = new src.type(T)
-		F.amount = amount
-		reagents.copy_to(F, (reagents.total_volume))
-		F.add_atom_colour(color, FIXED_COLOUR_PRIORITY)
-		F.metal = metal
-
-
-/obj/effect/particle_effect/foam/temperature_expose(datum/gas_mixture/air, exposed_temperature, exposed_volume)
-	if(prob(max(0, exposed_temperature - 475))) //foam dissolves when heated
-		kill_foam()
```
Comment (DaedalusGame) [inline code/game/objects/effects/effect_system/effects_foam.dm]: This is a behavior change that isn't mentioned in your changelog

File: code/game/objects/effects/effect_system/effects_foam.dm (hunk 1 above)
Comment (Iamgoofball) [reply code/game/objects/effects/effect_system/effects_foam.dm]: i removed fire melting foam to begin with

File: code/game/objects/effects/effect_system/effects_foam.dm (hunk 3 above)
Comment (Iamgoofball) [reply code/game/objects/effects/effect_system/effects_foam.dm]: ty for reminding me

**Comment by arsserpentarium** [comment]
Can foam and smoke be at one tile at the same time?

**Comment by DaedalusGame** [comment]
yes duh

**Comment by Tlaltecuhtli** [comment]
does this make smoke vapour instead of touch

**Comment by ExcessiveUseOfCobblestone** [comment]
Foam being touch will bring foamestry back unless that's the underlying intention here which is why I am very much into this pr.

**Comment by DaedalusGame** [comment]
this will be even worse than 4x foam but you can read the pr yourself to find out why

**Comment by ExcessiveUseOfCobblestone** [comment]
I wish you didn't direct my eyes to such a sentence @DaedalusGame 

Now I want this pr even more!!

**Comment by Iamgoofball** [comment]
Foamestry is bae

**Comment by duncathan** [comment]
fix your shit goofball

**Pull request closed without merging**